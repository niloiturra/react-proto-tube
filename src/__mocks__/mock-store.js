import { createStore, applyMiddleware } from 'redux';
import { logger } from 'redux-logger';
import createSagaMiddleware from 'redux-saga';

import reducers from '../redux/modules';
import rootSaga from '../redux/sagas';

export function createTestStore() {
  const saga = createSagaMiddleware();
  const middlewares = [saga, logger];
  const store = createStore(reducers, applyMiddleware(...middlewares));
  saga.run(rootSaga);

  return store;
}
