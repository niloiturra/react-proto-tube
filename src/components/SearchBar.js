import React from 'react';
import { Segment, Input } from 'semantic-ui-react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { searchVideoAsync } from '../redux/modules/search-video';

const SearchBar = (props) => {
  const getValueTerm = (e) => {
    if (e.keyCode === 13) {
      const valueTerm = e.target.value;
      props.searchVideoAsync(valueTerm);
    }
  };

  return (
    <div className="search-bar">
      <Segment stacked>
        <Input
          icon="search"
          onKeyDown={(e) => getValueTerm(e)}
          size="large"
          placeholder="React hot videos ðŸ˜³"
        />
      </Segment>
    </div>
  );
};

const mapDispatchToProps = (dispatch) =>
  bindActionCreators({ searchVideoAsync }, dispatch);

export default connect(null, mapDispatchToProps)(SearchBar);
