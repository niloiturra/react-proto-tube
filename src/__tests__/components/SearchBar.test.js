import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import SearchBar from '../../components/SearchBar';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
// eslint-disable-next-line jest/no-mocks-import
import { createTestStore } from '../../__mocks__/mock-store';

Enzyme.configure({ adapter: new Adapter() });

let store;
describe('testing SearchBar component render', () => {
  beforeEach(() => {
    store = createTestStore();
  });

  it('should render SearchBar component correctly', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <SearchBar />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
