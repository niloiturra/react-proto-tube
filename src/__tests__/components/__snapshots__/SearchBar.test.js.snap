// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing SearchBar component render should render SearchBar component correctly 1`] = `ShallowWrapper {}`;
